anisoap/representations/radial_basis.py

```diff
@@ -1,4 +1,90 @@
+import warnings
+
 import numpy as np
+import scipy.linalg
+from equistore.core import TensorMap
+from scipy.special import gamma
+
+
+def inverse_matrix_sqrt(matrix: np.array):
+    """
+    Returns the inverse matrix square root.
+    The inverse square root of the overlap matrix (or slices of the overlap matrix) yields the
+    orthonormalization matrix
+    Args:
+        matrix: np.array
+            Symmetric square matrix to find the inverse square root of
+
+    Returns:
+        inverse_sqrt_matrix: S^{-1/2}
+
+    """
+    if not np.allclose(matrix, matrix.conjugate().T):
+        raise ValueError("Matrix is not hermitian")
+    eva, eve = np.linalg.eigh(matrix)
+
+    if (eva < 0).any():
+        raise ValueError(
+            "Matrix is not positive semidefinite. Check that a valid gram matrix is passed."
+        )
+    return eve @ np.diag(1 / np.sqrt(eva)) @ eve.T
+
+
+def gto_square_norm(n, sigma):
+    """
+    Compute the square norm of GTOs (inner product of itself over R^3).
+    An unnormalized GTO of order n is \phi_n = r^n * e^{-r^2/(2*\sigma^2)}
+    The square norm of the unnormalized GTO has an analytic solution:
+    <\phi_n | \phi_n> = \int_0^\infty dr r^2 |\phi_n|^2 = 1/2 * \sigma^{2n+3} * \Gamma(n+3/2)
+    Args:
+        n: order of the GTO
+        sigma: width of the GTO
+
+    Returns:
+        square norm: The square norm of the unnormalized GTO
+    """
+    return 0.5 * sigma ** (2 * n + 3) * gamma(n + 1.5)
+
+
+def gto_prefactor(n, sigma):
+    """
+    Computes the normalization prefactor of an unnormalized GTO.
+    This prefactor is simply 1/sqrt(square_norm_area).
+    Scaling a GTO by this prefactor will ensure that the GTO has square norm equal to 1.
+    Args:
+        n: order of GTO
+        sigma: width of GTO
+
+    Returns:
+        N: normalization constant
+
+    """
+    return np.sqrt(1 / gto_square_norm(n, sigma))
+
+
+def gto_overlap(n, m, sigma_n, sigma_m):
+    """
+    Compute overlap of two *normalized* GTOs
+    Note that the overlap of two GTOs can be modeled as the square norm of one GTO, with an effective
+    n and sigma. All we need to do is to calculate those effective parameters, then compute the normalization.
+    <\phi_n, \phi_m> = \int_0^\infty dr r^2 r^n * e^{-r^2/(2*\sigma_n^2) * r^m * e^{-r^2/(2*\sigma_m^2)
+    = \int_0^\infty dr r^2 |r^{(n+m)/2} * e^{-r^2/4 * (1/\sigma_n^2 + 1/\sigma_m^2)}|^2
+    = \int_0^\infty dr r^2 r^n_{eff} * e^{-r^2/(2*\sigma_{eff}^2)
+    prefactor.
+    ---Arguments---
+    n: order of the first GTO
+    m: order of the second GTO
+    sigma_n: sigma parameter of the first GTO
+    sigma_m: sigma parameter of the second GTO
+
+    ---Returns---
+    S: overlap of the two normalized GTOs
+    """
+    N_n = gto_prefactor(n, sigma_n)
+    N_m = gto_prefactor(m, sigma_m)
+    n_eff = (n + m) / 2
+    sigma_eff = np.sqrt(2 * sigma_n**2 * sigma_m**2 / (sigma_n**2 + sigma_m**2))
+    return N_n * N_m * gto_square_norm(n_eff, sigma_eff)
```

Comment:
I agree -- this framework is general enough to handle different radial bases if we decide to implement them, in which case it would make more sense to move all this orthonormalization logic to a new utils. However, since we only have GTOs for now, I think making a new file containing this logic may be unnecessary indirection at the moment.